homeassistant/components/edl21/config_flow.py

```diff
@@ -0,0 +1,68 @@
+"""Config flow for EDL21 integration."""
+from typing import Any
+
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_NAME
+from homeassistant.data_entry_flow import FlowResult
+
+from .const import CONF_SERIAL_PORT, DEFAULT_DEVICE_NAME, DOMAIN
+
+
+class EDL21ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """EDL21 config flow."""
+
+    VERSION = 1
+
+    async def async_step_import(self, import_config: dict[str, Any]) -> FlowResult:
+        """Import a config entry from configuration.yaml."""
+
+        self._async_abort_entries_match(
+            {
+                CONF_SERIAL_PORT: import_config[CONF_SERIAL_PORT],
+            }
+        )
+        if import_config[CONF_NAME] != "":
+            self._async_abort_entries_match(
+                {
+                    CONF_NAME: import_config[CONF_NAME],
+                }
+            )
```

Comment:
Why bother with this check? I think it's wrong to have two entries with same port, but same name shouldn't be restricted I think.

Reply:
The name is used as part of the entities names here: https://github.com/home-assistant/core/blob/dev/homeassistant/components/edl21/sensor.py#L341
Configuring two smart meters with the same name would result in 'duplicate' entities.

Reply:
Since the entities have unique id, this shouldn't matter, so I'm still unconvinced by this check
Anyway you can clean it up like this:
```suggestion
        if import_config[CONF_NAME]:
            self._async_abort_entries_match(
                {CONF_NAME: import_config[CONF_NAME]}
            )
```

Reply:
Cleaned up as suggested.